ofrak_core/ofrak/core/esp.py

```diff
@@ -0,0 +1,779 @@
+                        version=struct.unpack("<I", f.read(4))[0],
+                        signature=f.read(64),
+                    )
+                    signature_offset = hash_offset + 32
+                    await resource.create_child_from_view(
+                        signature, data_range=Range.from_size(signature_offset, 80)
+                    )
+
+                elif image.secure_pad == "2":  # Secure Boot V2
+                    # TODO: ESPTool.py comment says: "after checksum: SHA-256 digest +
+                    # signature sector, but we place signature sector after the 64KB
+                    # boundary" so this might work but unsure
+                    f.seek(-64004, os.SEEK_END)
+                    signature = ESPSignature(
+                        version=struct.unpack("<I", f.read(4))[0],
+                        signature=f.read(64000),
+                    )
+                    signature_offset = hash_offset + 32
+                    await resource.create_child_from_view(
+                        signature, data_range=Range.from_size(signature_offset, 64004)
+                    )
+            # Process application description if present
+            if app_desc:
+                app_description, magic_word = self.__parse_app_description(app_desc)
+                if magic_word == 0xABCD5432:
+                    await resource.create_child_from_view(
+                        app_description,
+                        data_range=Range.from_size(
+                            ESP_HEADER_SIZE + ESP_EXTENDED_HEADER_SIZE + ESP_SEGMENT_HEADER_SIZE,
+                            256,
+                        ),
+                    )
+            elif bootloader_desc:
+                bootloader_description, magic_byte = self.__parse_bootloader_description(
+                    bootloader_desc
+                )
+                if magic_byte == 80:
+                    await resource.create_child_from_view(
+                        bootloader_description,
+                        data_range=Range.from_size(
+                            ESP_HEADER_SIZE + ESP_EXTENDED_HEADER_SIZE + ESP_SEGMENT_HEADER_SIZE, 80
+                        ),
+                    )
+
+    def __parse_flash_bits(self, image) -> Tuple[int, int]:
+        """
+        Extracts and returns flash size and frequency bits from the firmware image.
+
+        :param image: The firmware image object
+        :return: A tuple containing the flash size bits and flash frequency bits
+        """
+        flash_s_bits = image.flash_size_freq & 0xF0
+        flash_fr_bits = image.flash_size_freq & 0x0F
+        return flash_s_bits, flash_fr_bits
+
+    # TODO: Display boolean or value?
+    def __parse_wp_pin(self, image, chip):
+        if chip != "esp8266":
+            return False if image.wp_pin == image.WP_PIN_DISABLED else True
+        return None
+
+    # TODO: Probably useful to display chip name but not sure where to put it atm
```

Comment:
For things like this, it makes sense to add attributes to the `ESP` class and split the logic to parse these out into an analyzer. Sometimes there is overlap with the unpacking logic, and that's okay: the unpacker can call the analyzer and the code doesn't need to be duplicated.